lib/generate.js

```diff
@@ -135,6 +136,40 @@
         "id": "id"
     }
 
+    generate.js = generate.jstestfuzz = function (parser, exportName) {
+        var output = "// Generated automatically by nearley, version " + parser.version + "\n";
+        output +=  "// http://github.com/Hardmath123/nearley\n";
+        output += "(function () {\n"
+        output += "var compileGrammar = function (options) {\n";
+        output += "function id(x) { return x[0]; }\n";
+        output += parser.body.join('\n');
+        output += "var grammar = {\n";
+        output += "    Lexer: " + parser.config.lexer + ",\n";
+        output += "    ParserRules: " +
+            serializeRules(parser.rules, generate.javascript.builtinPostprocessors)
+            + "\n";
+        output += "  , ParserStart: " + JSON.stringify(parser.start) + "\n";
+        output += "}\n";
+        output += "return grammar;\n";
+        output += "}\n";
+        output += "if (typeof module !== 'undefined'"
+            + "&& typeof module.exports !== 'undefined') {\n";
+        output += "   module.exports = {compileGrammar};\n";
+        output += "} else {\n";
+        output += "   window." + exportName + " = {compileGrammar};\n";
+        output += "}\n";
+        output += "})();\n";
+        return output;
+    };
+
+    generate.jstestfuzz.builtinPostprocessors = {
+        "joiner": "function joiner(d) {return d.join('');}",
```

Comment:
I think you can use es6 here? This will be executed in the same environment as the compiled grammar.